Makefile

```diff
@@ -22,29 +21,51 @@ DEFAULT_IMAGES = ddev-php-base ddev-php-prod
 VERSION := $(shell git describe --tags --always --dirty)
 BUILDINFO = $(shell echo hash=$$(git rev-parse --short HEAD) Built $$(date) by $${USER} on $$(hostname) $(BUILD_IMAGE) )
 
+# In CI environments, use the plain Docker build progress to not overload the CI logs
+PROGRESS := $(if $(CI),plain,auto)
+
 #
 # This version-strategy uses a manual value to set the version string
 #VERSION := 1.2.3
 
-DOCKER_BUILDKIT=1
-
 build: images
 
 images: $(DEFAULT_IMAGES)
 
-push: images
-	for item in $(DEFAULT_IMAGES); do docker push $(DOCKER_ORG)/$$item:$(VERSION); echo "pushed $(DOCKER_ORG)/$$item"; done
+$(DEFAULT_IMAGES): prep .docker-build-info.txt
+	set -eu -o pipefail; \
+	DOCKER_BUILDKIT=1 docker buildx build --progress=$(PROGRESS) $(BUILDPUSHARG) --platform linux/amd64 --label com.ddev.buildhost=${shell hostname} --target=$@  -t $(DOCKER_ORG)/$@:$(VERSION) $(DOCKER_ARGS) .
 
-ddev-php-prod ddev-php-base: buildinfo
-	DOCKER_BUILDKIT=$(DOCKER_BUILDKIT) docker build --label com.ddev.buildhost=${shell hostname} --target=$@  -t $(DOCKER_ORG)/$@:$(VERSION) $(DOCKER_ARGS) .
+push: prep images multi_arch
+	set -eu -o pipefail; \
+	for item in $(DEFAULT_IMAGES); do \
+		docker buildx build --push --platform $(BUILD_ARCHS) --label com.ddev.buildhost=${shell hostname} --target=$$item  -t $(DOCKER_ORG)/$$item:$(VERSION) $(DOCKER_ARGS) .; \
+		echo "pushed $(DOCKER_ORG)/$$item:$(VERSION)"; \
+	done
 
-test: images
+multi_arch: prep
+	set -eu -o pipefail; \
+	for item in $(DEFAULT_IMAGES); do \
+		docker buildx build --platform $(BUILD_ARCHS) --label com.ddev.buildhost=${shell hostname} --target=$$item  -t $(DOCKER_ORG)/$$item:$(VERSION) $(DOCKER_ARGS) .; \
+		echo "created multi-arch builds $(DOCKER_ORG)/$$item:$(VERSION)"; \
+	done
+
+ddev-webserver:
+	docker buildx build --platform linux/amd64 -o type=docker --label com.ddev.buildhost=${shell hostname} --target=$@  -t $(DOCKER_ORG)/$@:$(VERSION) $(DOCKER_ARGS) .
+
+test: $(DEFAULT_IMAGES)
 	for item in $(DEFAULT_IMAGES); do \
 		if [ -x tests/$$item/test.sh ]; then tests/$$item/test.sh $(DOCKER_ORG)/$$item:$(VERSION); fi; \
 	done
 
 version:
 	@echo VERSION:$(VERSION)
 
-buildinfo:
+.docker-build-info.txt:
 	@echo "$(VERSION) $(BUILDINFO)" >.docker-build-info.txt
+
+prep:
```

Comment:
Assuming you're trying to build from Linux locally? Users of GitHub Desktop (Windows + Mac) don't need this step, that's why I only added it to the CircleCI config previously. But if this is needed to get it to run properly, then by all means 😄

Reply:
I'm using macOS edge (stable versions don't have buildx) and it definitely needed this.